actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -112,27 +114,84 @@ private void addCourseDetails(Request request, Map<String, Object> userCoursesRe
             requestBody,
             (Map<String, String>) request.getRequest().get(JsonKey.HEADER));
 
-    mergeDetailsAndSendCourses(contents, batches);
+    Map<String, Object> courseBatchesMap = null;
+    List<String> requestedFields = null;
+    String[] queryParams = (String[]) request.getContext().get(JsonKey.FIELDS);
+    requestedFields = new ArrayList<>(Arrays.asList(queryParams));
+
+    if (!CollectionUtils.isEmpty(requestedFields)) {
```

Comment:
Fetch course batch only if there is at least one supported field mentioned in requestedFields.